File: src/main/java/employees/java17/Employee.java
package employees.java17;

import java.math.BigDecimal;

record  Employee(String name, String surname,
                 Position position,
                 BigDecimal salaryPerHour) {


    public String getFullName() {
        return name + " " + surname;
    }

    @Override
    public String toString() {
        return getFullName() + " is working as a " + position.getDesc() + " with salary per hour: " + salaryPerHour.toPlainString() + " PLN";
    }
}

File: src/main/kotlin/employees/Employee.kt
package employees

import java.math.BigDecimal

data class Employee(private val name: String,
                    private val surname: String,
                    val position: Position,
                    val salaryPerHour: BigDecimal
) {

    val fullName = "$name $surname"

    override fun toString() = "$fullName is working as a ${position.positionName} with salary per hour: $salaryPerHour PLN"

}


